docs/source/guides/troubleshooting.rst

```diff
@@ -0,0 +1,65 @@
+
+Installing some packages on Ubuntu
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your repository provides outdated versions of tools - install them from source code.
+
+- CMake
+   Before installation make sure to get all dependencies:
+
+   - make
+   - curl (libcurl3-dev)
+   - zlib (zlib1g)
+
+  .. code:: sh
+
+    git clone https://gitlab.kitware.com/cmake/cmake.git /tmp/cmake;
+    (cd /tmp/cmake ; /tmp/cmake/bootstrap --system-curl --parallel=4 --enable-ccache);
+    make -j4 -C /tmp/cmake;
```

Comment:
I suggest remove parallelism options ``-j4`` from listings since we do not know anything about build environment, which will be used.